packages/cli/commands/config/set.js

```diff
@@ -1,25 +1,93 @@
-const { addConfigOptions, addAccountOptions } = require('../../lib/commonOpts');
-const defaultAccount = require('./set/defaultAccount');
-const defaultMode = require('./set/defaultMode');
-const httpTimeout = require('./set/httpTimeout');
-const allowUsageTracking = require('./set/allowUsageTracking');
+const { loadAndValidateOptions } = require('../../lib/validation');
 const { i18n } = require('@hubspot/cli-lib/lib/lang');
+const { getAccountId } = require('../../lib/commonOpts');
+const { trackCommandUsage } = require('../../lib/usageTracking');
+const { promptUser } = require('../../lib/prompts/promptUtils');
+const { EXIT_CODES } = require('../../lib/enums/exitCodes');
+const { setDefaultMode } = require('./set/defaultMode');
+const { setHttpTimeout } = require('./set/httpTimeout');
+const { setAllowUsageTracking } = require('./set/allowUsageTracking');
 
 const i18nKey = 'cli.commands.config.subcommands.set';
 
 exports.command = 'set';
 exports.describe = i18n(`${i18nKey}.describe`);
 
-exports.builder = yargs => {
-  addConfigOptions(yargs, true);
-  addAccountOptions(yargs, true);
+const selectOptions = async () => {
+  const { mode } = await promptUser([
+    {
+      type: 'list',
+      look: false,
+      name: 'mode',
+      pageSize: 20,
+      message: i18n(`${i18nKey}.promptMessage`),
+      choices: [
+        { name: 'Default mode', value: { defaultMode: '' } },
+        { name: 'Allow usage tracking', value: { allowUsageTracking: '' } },
+        { name: 'HTTP timeout', value: { httpTimeout: '' } },
+      ],
+    },
+  ]);
+
+  return mode;
+};
+
+const handleConfigUpdate = async (accountId, options) => {
+  const { defaultMode, httpTimeout, allowUsageTracking } = options;
+
+  if (typeof defaultMode !== 'undefined') {
+    await setDefaultMode({ defaultMode, accountId });
+    return true;
+  } else if (typeof httpTimeout !== 'undefined') {
+    await setHttpTimeout({ httpTimeout, accountId });
+    return true;
+  } else if (typeof allowUsageTracking !== 'undefined') {
+    await setAllowUsageTracking({ allowUsageTracking, accountId });
+    return true;
+  }
+
+  return false;
+};
+
+exports.handler = async options => {
+  await loadAndValidateOptions(options);
+
+  const accountId = getAccountId(options);
 
+  trackCommandUsage('config-set', {}, accountId);
+
+  const configUpdated = await handleConfigUpdate(accountId, options);
+
+  if (!configUpdated) {
+    const selectedOptions = await selectOptions();
+
+    await handleConfigUpdate(accountId, selectedOptions);
+  }
+
+  process.exit(EXIT_CODES.SUCCESS);
+};
+
+exports.builder = yargs => {
   yargs
-    .command(defaultAccount)
-    .command(defaultMode)
-    .command(httpTimeout)
-    .command(allowUsageTracking)
```

Comment:
General q: are you planning to bump the major version in the next release since we're removing commands here?

Reply:
We talked offline, but the tldr is that I probably won't bump a major version here because this command sees very low usage. Also, it very likely is not being utilized in any automation flows.

Reply:
I think we need to bump the major version for the `hs upload` command change.

Reply:
Are you referencing [this update](https://github.com/HubSpot/hubspot-cli/pull/666)? If so, I believe Anthony plans to make some updates to that PR to make it a non-breaking change.

Reply:
Sorry for not being more specific. I was thinking about #665. My thought process was that since it changes the error handling of the upload command that it warranted bumping the major version. For most CLI users, the process of upgrading is the same since it always involves running a command. The difference between major, minor, and patch only really comes into play when one of the CLI NPM packages is referenced in `package.json` or is a dep of another package.